docs/environment-variables.md

```diff
@@ -0,0 +1,170 @@
+## ENVIRONMENT VARIABLES
+#### The following environment variables can be set to configure XL Docker image for different scenarios
+
+### Common for all XL Products docker images,
+
+##### `APP_ROOT   ????`
+- value: "/opt/xebialabs"
+- Note: 
+    - not valid anymore as APP_ROOT is set to /opt/xebialabs .. however in db-drivers.sh if you set APP_ROOT to different path it will take affect but then scrib will be broken and will fail with below,
+cp: cannot stat '/opt//db-libs/h2*': No such file or directory
+and this will cause docker container to stop as DB libs can not be found.
+
+
+##### `APP_HOME  ????`
+- value: "/${APP_ROOT}/xl-deploy-server"
+- Note: 
+    - not valid anymore as APP_HOME is set to ${APP_ROOT}/{{ product }}-server .. if we tried to pass it as env docker container start will fail with below,
+/opt/xebialabs/xl-deploy-server/bin/run-in-container.sh: line 136: cd: /opt/default-plugins: No such file or directory
+
+##### `ADMIN_PASSWORD`
+- value: "password-value"
+- Admin user password used in XL Products
+- Note: 
+    - If this value is not passed a random 8 characters password will be generated using "pwgen 8"
+
+##### `REPOSITORY_KEYSTORE_PASSPHRASE`
+value: "passphrase-string" 
+Repository keystore passphrase used in XL Products
+If this value is not passed a random 16 characters password will be generated using "pwgen 8"
+Note: 
+    - Keystore password must be at least 6 characters
+
+##### `REPOSITORY_KEYSTORE`
+- value: "keystore-string"
+- Repository keystore passphrase used in XL Products
+- Note:    
+    - If this value is set then it will be converted to base64 value and added to keystore file like below,
+echo "${REPOSITORY_KEYSTORE}" | base64 -d > ${APP_HOME}/conf/repository-keystore.jceks
+    - If this value is not set then new one will be created using keystore passphrase like below,
+keytool -genseckey -alias deployit-passsword-key -keyalg aes -keysize 128 -keypass "deployit" -keystore ${APP_HOME}/conf/repository-keystore.jceks -storetype jceks -storepass ${REPOSITORY_KEYSTORE_PASSPHRASE}
+
+##### `XL_LICENSE`
```

Comment:
same for the license , need an example here. the license can be provided as a base64 string or mounted as part of conf volume. user has no idea what ${APP_HOME}/conf/xl-release-license.lic is

Reply:
Done and example updated .. mounting volumes will leave it for the other doc talking about it